### PR TITLE
Improving Gatekeeper navigation

### PR DESCRIPTION
**Pull request by codyrancher**
- Got rid of the old template list page and just use the generic now
 - Added a new constraints/_resource/_id route so navigation will activate appropriately.

 rancher/dashboard#1529

### CODE REVIEW

**Review by vincent99 — COMMENTED** [review]
The type in the header now says "K8sAlowedRepo" (or whatever you pick) instead of "Constraint".

I don't know what it is about this particular dropdown but this one (and only this one AFAICT) triggers the browser autocomplete and makes it unusable:

image

The entire contents of the tabs is completely unpolished and poorly formatted, spaced, aligned...

image

File: config/product/gatekeeper.js [hunk 1]
```diff
@@ -44,17 +46,20 @@ export function init(store) {
     label:      'Constraint',
     namespaced: false,
     name:       'gatekeeper-constraint',
+    ifHaveType: 'constraints.gatekeeper.sh.k8sallowedrepos',
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply config/product/gatekeeper.js:49]: That was a test that should've been removed.

**Comment by codyrancher** [comment]
> The type in the header now says "K8sAlowedRepo" (or whatever you pick) instead of "Constraint".

Fixed

> I don't know what it is about this particular dropdown but this one (and only this one AFAICT) triggers the browser autocomplete and makes it unusable:

I can't reproduce and autocomplete is already turned off
Screen Shot 2020-10-20 at 3 38 47 PM

> The entire contents of the tabs is completely unpolished and poorly formatted, spaced, aligned...

There have been changes made by others which have broken this page. I only tested what I thought was relevant to the issue and the changes I've made which didn't include testing the tabs. I think either we need to write tests to catch these regressions (though I'd probably discourage stylistic tests) or maybe we should stop using slots within things like ArrayList and KeyValue ( I probably did most of that). Do you have any thoughts on that?

Anyway, even though I don't consider the stylistic changes relevant to the pr I've made some changes which hopefully you're happy with.
Screen Shot 2020-10-20 at 3 58 59 PM
Screen Shot 2020-10-20 at 3 58 54 PM
Screen Shot 2020-10-20 at 3 58 45 PM

**Comment by vincent99** [comment]
I mean that's better and fixes some of the most obviously bad things, but still plenty wrong with this form.  We can deal with it later I guess...

**Pull request merged**